src/Data.Json.JSemantic.purs

```diff
@@ -0,0 +1,105 @@
+module Data.Json.JSemantic
+  ( JSemantic(..)
+  , toSemantic
+  ) where
+
+import Data.Argonaut.JCursor(JsonPrim(..), runJsonPrim)
+import Data.Maybe
+import Data.String.Regex (RegexFlags(..), regex, test)
+import Data.String
+import Data.Foldable
+import qualified Data.Array as A
+import qualified Data.Array.Unsafe as AU
+import Data.Tuple
+import qualified Data.Date as Date
+import Math
+
+
+data JSemantic = Integral   Number
+               | Fractional Number
+               | Date       Date.Date
+               | DateTime   Date.Date
+               | Time       Date.Date
+               | Interval   Date.Date Date.Date
```

Comment:
I'm not sure we can represent an interval using two dates. We really need a good `purescript-duration` library, but no such thing exists (although there's `purescript-moment`, and I know the underlying JS library [has support for durations](http://momentjs.com/docs/#/durations/)). ISO8601 defines the string representation for them. 

[There are some terrifying](https://gist.github.com/philipashlock/8830168) regexes to parse duration, but not sure they'd be that useful, because the renderer would need a typed ADT in order to provide a useful styling of things like "every other week".

Not sure if we want to address this in this PR unless there's a quickfix, otherwise I'm fine ignoring intervals for now. Can always add them later after we have a good PS duration library.